Add VLM input formats tests

JustinTong0323 opened the pull request:
<!-- Thank you for your contribution! We appreciate it. The following guidelines will help improve your pull request and facilitate feedback. If anything is unclear, don't hesitate to submit your pull request and ask the maintainers for assistance. -->

## Motivation

This PR is a Supplement for https://github.com/sgl-project/sglang/pull/6136

Add tests for passing pixel values and precomputed features to VLMs

Split from test_vlm_acc

<!-- Explain the purpose of this PR and the goals it aims to achieve. -->

## Modifications

<!-- Describe the changes made in this PR. -->

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

JustinTong0323 committed:
Refactor test_vlm_acc for better extensibility

Signed-off-by: Xinyuan Tong <justinning0323@outlook.com>

JustinTong0323 committed:
fix qwen_vl

Signed-off-by: Xinyuan Tong <justinning0323@outlook.com>

JustinTong0323 committed:
fix qwen

Signed-off-by: Xinyuan Tong <justinning0323@outlook.com>

JustinTong0323 committed:
Merge branch 'main' into refactor_test_vlm_accuracy

JustinTong0323 committed:
split the test

Signed-off-by: Xinyuan Tong <justinning0323@outlook.com>

JustinTong0323 committed:
minor

Signed-off-by: Xinyuan Tong <justinning0323@outlook.com>

JustinTong0323 committed:
Replace test_vlm_acc in run_suite

Signed-off-by: Xinyuan Tong <justinning0323@outlook.com>

JustinTong0323 committed:
Merge branch 'main' into refactor_test_vlm_accuracy

JustinTong0323 committed:
Merge branch 'main' into refactor_test_vlm_accuracy

mickqian commented on test/srt/test_vlm_input_format.py:
```diff
@@ -0,0 +1,186 @@
+        )
+        cls.vision_tower = model.vision_tower.eval().to(cls.device)
+        cls.mm_projector = model.multi_modal_projector.eval().to(cls.device)
+        cls.visual = lambda processor_output: cls.mm_projector(
```
Does this means we need to customize the test for different vlms?

JustinTong0323 replied:
If I understand your question correctly, then yes, the visual components for each Visual Language Model (VLM) are distinct, and thus, we need to make adjustments for each individual VLM.

JustinTong0323 commented on test/srt/test_vlm_input_format.py:
```diff
@@ -0,0 +1,186 @@
+        cls._init_visual()
+
+    @classmethod
+    # @abstractmethod
```
ah yeah, forget to delete

JustinTong0323 committed:
refactor: Update VLMInputTestBase methods for clarity and implementation requirements

- Removed abstract method decorator from _init_visual and clarified its purpose.
- Updated _pixel_values_image_data to raise NotImplementedError, indicating it should be overridden in subclasses.
- Cleaned up comments for better understanding of method requirements.

Signed-off-by: Xinyuan Tong <justinning0323@outlook.com>

JustinTong0323 committed:
Merge branch 'main' into refactor_test_vlm_accuracy

JustinTong0323 commented:
merge into #6543
